src/Spatial/Internals/ImmutableList.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace MathNet.Spatial.Internals
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    internal sealed class ImmutableList<T> : IEnumerable<T>
+    {
+        internal static readonly ImmutableList<T> Empty = new ImmutableList<T>();
+
+        private readonly T[] data;
+
+        private ImmutableList()
+        {
+            this.data = new T[0];
+        }
+
+        internal ImmutableList(T[] data)
```

Comment:
so... we're relying that the user doesn't keep the argument around? worth documenting if that's the case

Reply:
Yes, maybe better to make this private and have the internal ctor take an `IEnumerable<T>` and ToArray it.